src/object/omitBy.ts

```diff
@@ -22,12 +22,10 @@ export function omitBy<T extends Record<string, any>>(
   shouldOmit: (value: T[keyof T], key: keyof T) => boolean
 ): Partial<T> {
   const result: Partial<T> = {};
-
-  const objEntries = Object.entries(obj);
-  for (let i = 0; i < objEntries.length; i++) {
-    const [key, value] = objEntries[i];
+  for (const key in obj) {
+    const value = obj[key];
     if (!shouldOmit(value, key)) {
-      (result as any)[key] = value;
+      result[key] = value;
```

Comment:
Change `Object.entries` to `for...in`. 
This also improves the `any` type assertion.